java/src/Ice/src/main/java/com/zeroc/IceInternal/Transceiver.java

```diff
@@ -21,6 +21,15 @@ public interface Transceiver {
 
   int read(Buffer buf);
 
+  /**
+   * Checks if this transceiver is waiting to be read, typically because it has bytes readily
+   * available for reading.
+   *
+   * @return true if this transceiver is waiting to be read, false otherwise.
+   * @remark The caller must ensure the transceiver is not closed when calling this method.
+   */
+  boolean isWaitingToBeRead();
```

Comment:
Why don't we have this method in C#, is it because of the missing fix for back pressure?

Reply:
Yes, we'll need to add it to C# after we merge the C# back-pressure support (recently added to the 3.7 branch).